webapp/app/sw.index.js

```diff
@@ -59,8 +59,36 @@ self.addEventListener('sync', (event) => {
   }
 });
 
+self.addEventListener('fetch', (event) => {
+  const { request } = event;
+
+  const response = caches
+    .match(request)
+    .then((response) => {
+      if (response) {
+        return response;
+      }
+
+      return fetch(request)
+        .then(resp => {
+          if (request.url.endsWith('.png')) {
+            return caches.open('v1').then(cache => {
+              cache.put(event.request, resp.clone());
+              return resp;
+            });
+          }
+        });
+    });
+
+  event.respondWith(response);
+});
+
 self.addEventListener('message', event => {
   if (EVENT_TYPES.indexOf(event.data.type) > -1) {
     drinkWaterHistory.push(event.data);
   }
 });
+
+// self.addEventListener('unregister', event => {
```

Comment:
możemy to wywalić?

Reply:
@jackowiak 
Chodzi jedynie o:

> You are responsible for implementing how your script (service worker) handles updates to the cache. All updates to items in the cache must be explicitly requested; items will not expire and must be deleted.

Więc w sumie kiedyś powinniśmy ten cache czyścić i tyle